### PR TITLE
Add Welcome to the Dev Team doc

### PR DESCRIPTION
**Pull request by Bilalscape12**
Adds the new welcome doc given to graduating jr. devs into the docs site.

### CODE REVIEW

File: docs/developer-docs/developer-welcome.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
+---
+title: Welcome to the Dev Team
+description: Welcomes newly graduated developers, teaches how to upload a set, and educates on all of the privileges exclusive to developers.
+---
+
+# Welcome to the Dev Team!
+
+Now that you’re a dev I'm sure the first thing you want to know is how you can upload your set and what other options and privileges are now open to you. You'll have plenty of fancy new features on the website that you'll notice when going to a game’s page as well as a few other places, which we will go over here.
```
Comment (wescopeland) [inline docs/developer-docs/developer-welcome.md]:
```suggestion
Now that you’re a developer, I'm sure the first thing you want to know is how you can upload your set and what other options and privileges are now open to you. You'll have plenty of fancy new features on the website that you'll notice when going to a game’s page as well as a few other places, which we will go over here.
```

File: docs/developer-docs/developer-welcome.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+Now that you’re a dev I'm sure the first thing you want to know is how you can upload your set and what other options and privileges are now open to you. You'll have plenty of fancy new features on the website that you'll notice when going to a game’s page as well as a few other places, which we will go over here.
+
+[[toc]]
+
+## Set Management
+
+The web team is working on a new Management page which is where all dev related activities will end up being moved to. It can be accessed through the “Manage” button right above the Dev dropdown menu.
+
+This page must be used for editing release dates as well as **drag-and-drop reordering**.
```
Comment (wescopeland) [inline docs/developer-docs/developer-welcome.md]:
```suggestion
This page is used for editing release dates as well as **drag-and-drop reordering**.
```
One thing to note: eventually _all_ of the game metadata editing will be on the page (it will be completely removed from the normal site). May just be worth a mental note for now.

File: docs/developer-docs/developer-welcome.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+![Example of a game page viewed through the Management site](/public/management-site.png)
+
+## Uploading your First Set!
+
+The quickest and easiest way to upload your first set is through the management page of your set. From there you can mark the achievements you want to promote and then click “Promote selected” to promote them all in a few clicks!
+
+To get to this page, go to your game’s page:
+1. Click on Manage
+2. Scroll down until you see the list of achievements
+3. If you’d like, you can reorder them by clicking Start Reordering and then dragging-and-dropping the achievements
```
Comment (wescopeland) [inline docs/developer-docs/developer-welcome.md]:
```suggestion
3. If you’d like, you can reorder them by clicking the Start Reordering button and then dragging-and-dropping the achievements
```

File: docs/developer-docs/developer-welcome.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+4. Check the achievements you’d like to promote (Be careful to exclude any unwanted or voided achievements if there are any, otherwise just hit “All”)
+5. Hit Bulk promote or demote
```
Comment (wescopeland) [inline docs/developer-docs/developer-welcome.md]:
```suggestion
5. Click Bulk promote or demote
```

File: docs/developer-docs/developer-welcome.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+6. Hit Promote selected
```
Comment (wescopeland) [inline docs/developer-docs/developer-welcome.md]:
```suggestion
6. Click Promote selected
```

File: docs/developer-docs/developer-welcome.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+![Achievements being promoted through the Management site](/public/management-achievement-promotion.png)
+
+## Achievement-News Announcement
+
+Now that your set is live it is time to announce it to everyone if you are on Discord!
```
Comment (wescopeland) [inline docs/developer-docs/developer-welcome.md]:
```suggestion
Now that your set is live, it is time to announce it to everyone if you are on Discord!
```

File: docs/developer-docs/developer-welcome.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+
+We have a handy bot command that you can use to auto generate a template *as long as your set is in core*. Go to #bot-spam and type in “!gan [gameid]” where [gameid] is the id of your game. 
+
+The pinned messages in #devs-chat has an alternate colored template that can be used as well.
```
Comment (wescopeland) [inline docs/developer-docs/developer-welcome.md]:
```suggestion
The pinned messages in #devs-chat have an alternate colored template that can be used as well.
```

File: docs/developer-docs/developer-welcome.md [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+
+Then you can copy the results and edit the @{AUTHOR_NAME} with your name and post it in the #achievement-news channel with an @Achievement-News ping to get everyone's attention on the hottest set that just dropped on RA.
```
Comment (wescopeland) [inline docs/developer-docs/developer-welcome.md]:
```suggestion
You can copy the results, replace the @{AUTHOR_NAME} with your name, and post it in the #achievement-news channel with an @Achievement-News ping to get everyone's attention on the hottest set that just dropped on RA.
```

File: docs/developer-docs/developer-welcome.md [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+
+## Claims
+
+Now that you are a developer, the claim limit has expanded to a total of 4 concurrent regular claims. As long as you have no active tickets, you are free to claim sets. For more specific details look over the [claims section of the Developer Guidelines](/guidelines/developers/claims-system).
+
+If the game does not yet have a page, see [Creating Game Pages and Linking Hashes](#creating-game-pages-and-linking-hashes).
```
Comment (wescopeland) [inline docs/developer-docs/developer-welcome.md]:
```suggestion
If the game does not yet have a page, see [Creating Game Pages and Linking Hashes](#creating-game-pages-and-linking-hashes) for guidance.
```

File: docs/developer-docs/developer-welcome.md [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+
+## Set Revisions
+
+Achievement set revisions are an important part in keeping older sets up-to-date with modern standards. This is your chance to ensure that quality is being maintained on the site. Votes are done through the #revision-voting and #revision-discussion channels of the Discord server. You will be given the @Revision-Voting and @Rescore roles by default upon graduation.
```
Comment (wescopeland) [inline docs/developer-docs/developer-welcome.md]:
```suggestion
Achievement set revisions are an important part in keeping older sets up-to-date with modern standards. This is your chance to ensure that quality is maintained on the site. Votes are done through the #revision-voting and #revision-discussion channels of the Discord server. You will be given the @Revision-Voting and @Rescore roles by default upon graduation.
```

File: docs/developer-docs/developer-welcome.md [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+
+When considering whether to vote to approve a revision proposal, please familiarize yourself with the revision goal criteria outlined in the discussion forum pinned post. The goal of a revision is to turn a lacking set into a high quality, modern set that ideally will never need another revision. Voting in revision proposals is an important responsibility that all developers should take seriously and participate in for the long term health of the Retroachievements project.
```
Comment (wescopeland) [inline docs/developer-docs/developer-welcome.md]:
```suggestion
When considering whether to vote to approve a revision proposal, please familiarize yourself with the revision goal criteria outlined in the discussion forum pinned post. The goal of a revision is to turn a lacking set into a high-quality, modern set that ideally will never need another revision. Voting in revision proposals is an important responsibility that all developers should take seriously and participate in for the long-term health of the RetroAchievements project.
```

File: docs/developer-docs/developer-welcome.md [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+
+When proposing a revision, it requires contacting all active authors of the relevant set. If you wish to Opt-Out of being contacted, use [this form](https://forms.gle/mgzv7RHbJEPCrxc77).
+
+For details on how to propose a revision, see [Achievement Set Revisions](/guidelines/content/achievement-set-revisions)
+
+In the case where you are re-authoring the set (keeping the old achievements in unofficial and promoting your new reworked achievements, make sure to contact DevCompliance for approval) and you want to transfer over the earners for some/all of the previous achievements then you will need to reach out to RAdmin or #admin-site-cleanup on Discord to help you with this.
```
Comment (wescopeland) [inline docs/developer-docs/developer-welcome.md]:
```suggestion
If you are re-authoring the set (keeping the old achievements in Unofficial and promoting your new reworked achievements), make sure to contact DevCompliance for approval. If you want to transfer the earners for some or all of the previous achievements, you will need to reach out to RAdmin or #admin-site-cleanup on Discord for assistance.
```

File: docs/developer-docs/developer-welcome.md [hunk 13]
```diff
@@ -0,0 +1,181 @@
...
+
+And so on...
+
+## Emulator Updates
+
+You will now have a few new privileges in the RA emulators to take advantage of.
+
+### Creating Game Pages and Linking Hashes
+
+The biggest one is that now you can link new hashes to games by loading the game in an RA emulator and linking it.
```
Comment (wescopeland) [inline docs/developer-docs/developer-welcome.md]:
```suggestion
The biggest new privilege is that you can now link new hashes to games by loading the game in an RA emulator and linking it.
```

File: docs/developer-docs/developer-welcome.md [hunk 14]
```diff
@@ -0,0 +1,181 @@
...
+
+### Achievements
+
+Developers can directly make edits to achievements in Core.
+
+You can now freely promote and demote any achievements from Unofficial and Core through the emulator but it is much easier to do this through the site.
+
+## Site Updates
+
+You’ll now have a lot more access to tools on the site to help you as a developer.
```
Comment (wescopeland) [inline docs/developer-docs/developer-welcome.md]:
```suggestion
You now have access to many more tools on the site to help you as a developer.
```

File: docs/developer-docs/developer-welcome.md [hunk 15]
```diff
@@ -0,0 +1,181 @@
...
+
+![Want to Develop Games button](/public/want-to-develop.png)
+
+### Game Page
+
+#### Creating Forum Topics
+
+When you go to claim a set, you no longer need to create a forum topic separately. The Make Claim button will automatically create a forum topic for you if one does not exist. 
+
+Keep in mind that you should NOT create forum topics without reasons such as claiming a set, because then you will be the only person (besides admins) that can edit the top post which is usually where important information such as linked hashes or set notes are placed.
```
Comment (wescopeland) [inline docs/developer-docs/developer-welcome.md]:
```suggestion
Keep in mind that you should NOT create forum topics without a valid reason, such as claiming a set. Otherwise, you will be the only person (besides admins) who can edit the top post, which is usually where important information such as linked hashes or set notes is placed.
```

File: docs/developer-docs/developer-welcome.md [hunk 16]
```diff
@@ -0,0 +1,181 @@
...
+
+- **#ratools**
+
+- **#revision-voting**
+
+- **#revision-discussion**
+
+## Developer Events!
+
+There are a few events limited to developers which also have their own Discord channel
```
Comment (wescopeland) [inline docs/developer-docs/developer-welcome.md]:
```suggestion
There are a few events limited to developers, each with their own Discord channel:
```

**Review by wescopeland — COMMENTED** [review]
Looks good - most of the feedback here is just grammatical/spelling cleanup

**Comment by Bilalscape12** [comment]
The doc will need further revising when metadata editing gets moved from the normal site to the management site (since the dev menu will no longer exist), but for now the doc is fine as it is.

**Comment by wescopeland** [comment]
Looks good - only remaining thing I can think of is should this be linked from any of the sidenavs anywhere? At the moment I'm not sure anything directly links to this page. As a result, it won't be found by the search feature.

**Comment by redwizard42** [comment]
> Looks good - only remaining thing I can think of is should this be linked from any of the sidenavs anywhere? At the moment I'm not sure anything directly links to this page. As a result, it won't be found by the search feature.

My expectation is that we only link this upon a new dev joining since that's its usage; we may link it as a pin (or have it in the channel topic) in the dev channel.

Edit: Forgot which welcome doc this was. :X

**Review by redwizard42 — CHANGES_REQUESTED** [review]
Edit: Misread something, so deleting this comment.

File: docs/developer-docs/developer-welcome.md [hunk 17]
```diff
@@ -0,0 +1,181 @@
...
+
+When considering whether to vote to approve a revision proposal, please familiarize yourself with the revision goal criteria outlined in the discussion forum pinned post. The goal of a revision is to turn a lacking set into a high-quality, modern set that ideally will never need another revision. Voting in revision proposals is an important responsibility that all developers should take seriously and participate in for the long-term health of the RetroAchievements project.
+
+When proposing a revision, it requires contacting all active authors of the relevant set. If you wish to Opt-Out of being contacted, use [this form](https://forms.gle/mgzv7RHbJEPCrxc77).
+
+For details on how to propose a revision, see [Achievement Set Revisions](/guidelines/content/achievement-set-revisions)
+
+If you are re-authoring the set (keeping the old achievements in Unofficial and promoting your new reworked achievements), make sure to contact DevCompliance for approval. If you want to transfer the earners for some or all of the previous achievements, you will need to reach out to RAdmin or #admin-site-cleanup on Discord for assistance.
```
Comment (redwizard42) [inline docs/developer-docs/developer-welcome.md]: I would reword this to "If you are considering requesting a re-author of the set... ...contact DevCompliance for approval." and include a note to not put reauthoring in the revision proposal.

File: docs/developer-docs/developer-welcome.md [hunk 18]
```diff
@@ -0,0 +1,181 @@
...
+
+## Emulator Updates
+
+You will now have a few new privileges in the RA emulators to take advantage of.
+
+### Creating Game Pages and Linking Hashes
+
+The biggest new privilege is that you can now link new hashes to games by loading the game in an RA emulator and linking it.
+
+If creating a new page, select "New Title" and write an appropriate name in the text box. Check if there are any junk pages labeled ~Z~ which can be recycled.
```
Comment (redwizard42) [inline docs/developer-docs/developer-welcome.md]: Suggest update to "\~Z\~ No Hash"

File: docs/developer-docs/developer-welcome.md [hunk 19]
```diff
@@ -0,0 +1,181 @@
...
+
+### Code Notes
+
+You are now able to overwrite and delete any existing code notes. Be careful not to overwrite other’s notes without reason, if you are making improvements or correcting a note that is an acceptable reason.
+
+### Achievements
+
+Developers can directly make edits to achievements in Core.
+
+You can now freely promote and demote any achievements from Unofficial and Core through the emulator but it is much easier to do this through the site.
```
Comment (redwizard42) [inline docs/developer-docs/developer-welcome.md:110]: Consider adding a note that they no longer have to demote to unofficial to make changes and can publish changes directly in core.  Sometimes this is a new-dev confusion.

File: docs/developer-docs/developer-welcome.md [hunk 20]
```diff
@@ -0,0 +1,181 @@
...
+
+- **Subscribe to Tickets**: Useful if you decide to work on inactive developer sets and want to track any future tickets
+
+- **Want to Develop**: Adds/Removes the game on your Want to Develop list
+
+![Dev menu](/public/dev-menu.png)
+
+#### Adding Hubs and Related Games
+
+Hubs and related games can be added to game pages by entering the page IDs and clicking “Add”. Hubs managed by the QA and Art teams should not be added/removed, send a message to the team instead.
```
Comment (redwizard42) [inline docs/developer-docs/developer-welcome.md]: Add DevComp to the teams in this list since we have our own hubs as well that shouldn't be tampered with

**Review by redwizard42 — CHANGES_REQUESTED** [review]
A few changes I think would improve it. Otherwise looking good.

**Comment by Bilalscape12** [comment]
Changes made.
I agree that the page should not be linked since it is meant to be read exclusively by graduating juniors and could be potentially confusing for anyone else.

**Review by wescopeland — APPROVED** [review]

**Review by redwizard42 — APPROVED** [review]
